modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateTimeInput.es.js

```diff
@@ -60,36 +61,34 @@ class DateTimeInput extends React.Component {
 	};
 
 	_saveDateTimeValue = () => {
-		const date = dateFns.format(this.state.value, INPUT_DATE_FORMAT);
+		const dateInput = dateFns.format(this.state.value, INPUT_DATE_FORMAT);
 
-		if (date !== 'Invalid Date') {
-			this.setState(
-				{
-					value: date,
-				},
-				() => {
-					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(date, INPUT_DATE_FORMAT)
-							.toISOString(),
-					});
-				}
-			);
-		}
-		else {
-			const resetDate = dateFns.format(new Date(), INPUT_DATE_FORMAT);
+		if (this.state.previousValue !== dateInput) {
+			const newDateInput =
+				dateInput !== 'Invalid Date'
+					? dateInput
+					: dateFns.format(new Date(), INPUT_DATE_FORMAT);
 
 			this.setState(
 				{
-					value: resetDate,
+					previousValue: newDateInput,
+					value: newDateInput,
 				},
 				() => {
 					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(resetDate, INPUT_DATE_FORMAT)
-							.toISOString(),
+						type: this.props.propertyType,
+						value:
+							this.props.propertyType === 'date-time'
```

Comment:
It could be more elegant to change the string 'date-time' for PROPERTY_TYPES.DATE_TIME. This is just a suggestion :) .

Reply:
true, thanks